File: Tests/sample-exe/Builder.js
const ProjectBuilder = require("../ProjectBuilder");

module.exports = class Builder extends ProjectBuilder {
  executeBuild() {
    this.binary = `${this.buildPath}/${this.component}`;
    var cmd = `
      cd "${this.buildPath}" &&
      ${this.swftcCmdPath} -emit-executable -o "${this.binary}"
      "${this.sourcesPath}/main.swift"
    `;
    this.executeCommands(cmd);
    this.copyLibs();
    this.execute(`file ${this.binary}`);
  }
};

File: Tests/sample-exe/Builder.rb
require_relative "../Scripts/ProjectBuilder.rb"

# See:
# -  Build error: No such module "SwiftGlibc" – https://lists.swift.org/pipermail/swift-dev/Week-of-Mon-20160919/002955.html
class Builder < ProjectBuilder

   def initialize(arch)
      @root = File.dirname(__FILE__)
      super("hello-exe", arch)
   end

   def executeBuild
      binary = "#{@builds}/#{@component}"
      execute "cd #{@builds} && #{@swiftc} -emit-executable -o #{binary} #{@sources}/main.swift"
      execute "file #{binary}"
      copyLibs()
   end

end
